src/bvh/bvh_impl.rs

```diff
@@ -438,6 +439,20 @@ impl BVH {
             .collect::<Vec<_>>()
     }
 
+    /// Creates a [`BVHIterator`] to traverse the [`BVH`].
+    /// Returns a subset of `shapes`, in which the [`AABB`]s of the elements were hit by `ray`.
+    ///
+    /// [`BVH`]: struct.BVH.html
+    /// [`AABB`]: ../aabb/struct.AABB.html
+    ///
+    pub fn traverse_iterator<'a, Shape: Bounded>(
```

Comment:
Can you check whether anonymous lifetimes `'_` would be ok to use here? I don't think they were available back when this code was made. You don't have to change it here necessarily at this point but I'd just like to know whether it'd be an option as it'd make a great follow-up. :)

Reply:
I'm not 100% sure, but I don't think it applies to the new iterator code..

`impl<'a, Shape: Bounded> BVHTraverseIterator<'a, Shape> {`
needs a named lifetime parameter for
`pub fn new(bvh: &'a BVH, ray: &'a Ray, shapes: &'a [Shape]) -> Self`

And
`impl<'a, Shape: Bounded> Iterator for BVHTraverseIterator<'a, Shape> {`
needs it for
`type Item = &'a Shape;`

So I'm not sure if the anonymous syntax can be used anywhere. Unless I misunderstand the Rust 2018 docs.